Ensures each manufacturable class gets a different factory instance

Fito opened the pull request:
This fixes the following issue: https://github.com/entelo/industrialist/issues/8

aridlehoover commented on lib/industrialist/manufacturable.rb:
```diff
@@ -4,17 +4,20 @@ module Industrialist
   module Manufacturable
     def self.included(base)
       base.extend ClassMethods
+      base.class_variable_set(:@@factory, Industrialist::Factory.new)
```
So, I need to type out what this is doing so I understand it:
1. When instantiating a `Sedan`
2. Which inherits from `Automobile`
3. Which includes `Industrialist::Manufacturable`
4. The `included` method executes
5. The `Automobile` class is extended with the `create_factory`, `corresponds_to`, and `factory` methods
6. A class level variable called `@@factory` is set on the Automobile class

So, when complete, I have the following:
```ruby
class Automobile
  @@factory = Industrialist::Factory.new

  def self.create_factory(identifier)
    Object.const_set(identifier, factory)
  end

  def self.corresponds_to(key)
    factory.register(key, self)
  end

  def self.factory
    class_variable_get(:@@factory)
  end

  # plus whatever else is in the base Automobile class
end
```
Correct?

Questions:
* Why can't you just use `@@factory`? Why do you have to use `class_variable_set` and `class_variable_get`?

Fito replied:
1. Correct, except that `included` is only executed when `Automobile` is defined. `included` is not executed when defining or instantiating `Sedan`. `Sedan` just inherits all those methods from `Automobile` as if the methods had been defined there.
2. Using `@@factory` inside the `Manufacturable` module defines `@@factory` as a class variable for the `Manufacturable` module and not for the `Automobile` class. That means all classes that include `Manufacturable` share the same `@@factory` class variable. 
3. Calling `class_variable_set` on the base sets `@@factory` on `Automobile`, and `class_variable_get` gets it from `Automobile`.

aridlehoover replied:
So using `@factory` set an instance variable on the `Manufacturable` model? Or, in the `Automobile` when that class is defined? How'd that work at all?

Fito replied:
Yes. using `@factory` sets a class instance variable on `Manufacturable`. In that case, `Automobile` does not get an instance of `@factory`:
```ruby
module Industrialist
  module Manufacturable
    def self.included(base)
      base.extend ClassMethods
    end

    module ClassMethods
      @factory = Industrialist::Factory.new

      def create_factory(identifier)
        Object.const_set(identifier, @factory)
      end

      def corresponds_to(key)
        @factory.register(key, self)
      end
    end
  end
end

class Automobile
  include Industrialist::Manufacturable
  create_factory :AutomobileFactory
end

class Sedan < Automobile
  corresponds_to :sedan # => This throws an exception
end

`NoMethodError: undefined method `register' for nil:NilClass`
```
The problem is that in this case we are setting `@factory` on Manufacturable, and not on `Automobile`

So, if we use `instance_variable_set` to set `@factory` on the `Automobile` class, we get the following:
```ruby
module Industrialist
  module Manufacturable
    def self.included(base)
      base.extend ClassMethods
      base.instance_variable_set(:@factory, Industrialist::Factory.new)
    end

    module ClassMethods
      def create_factory(identifier)
        Object.const_set(identifier, @factory)
      end

      def corresponds_to(key)
        instance_variable_get(:@factory).register(key, self)
      end
    end
  end
end

class Automobile
  include Industrialist::Manufacturable
  create_factory :AutomobileFactory
end

class Sedan < Automobile
  corresponds_to :sedan # => This throws an exception
end

`NoMethodError: undefined method `register' for nil:NilClass`
```
We have the same problem. In this case `Automobile` has `@factory` defined, but `Sedan` can't see it. Class instance variables are not visible to children of the class:
```ruby
class Automobile
  @wheels = 4

  def self.get_wheels
    @wheels
  end
end

class Sedan < Automobile; end;

Automobile.get_wheels # => 4
Sedan.get_wheels #=> nil
```
On the other hand, class variables are visible to the children of the class that defines it:
```ruby
class Vehicle; end

class Automobile < Vehicle
  @@wheels = 4

  def self.get_wheels
    @@wheels
  end
end

class Sedan < Automobile; end;
class Book; end;

Automobile.get_wheels # => 4
Sedan.get_wheels #=> 4
Vehicle.class_variable_get(:@@wheels) #=> Throws an exception
`NameError: uninitialized class variable @@wheels in Vehicle`
```
Another way to achieve this is by tapping into the `inherited(base)` callback and manually assign the same factory to each children (This is the fix proposed in Jason's [PR](https://github.com/entelo/industrialist/pull/7)):
```ruby
module Industrialist
  module Manufacturable
    def self.included(base)
      base.extend ClassMethods
      base.instance_variable_set(:@factory, Industrialist::Factory.new)
    end

    module ClassMethods
      def inherited(base)
        base.instance_variable_set(:@factory, @factory)
      end

      def create_factory(identifier)
        Object.const_set(identifier, @factory)
      end

      def corresponds_to(key)
        @factory.register(key, self)
      end
    end
  end
end
```
Given that we want children of a manufacturable class to share the same factory, we either need to use `@@factory` or leverage `inherited` to assign the same factory instance on `@factory`.

I'm ok going with either of those. Thoughts?

Commit:
Ensures each manufacturable class gets a different factory instance

- This fixes the following issue: https://github.com/entelo/industrialist/issues/8
- Warns when including manufacturable twice in one class hierarchy
- Bumps gem to 0.2.1
